### PR TITLE
Feat: Login.gov help modal

### LINKED ISSUE #1473: Eligibility Index - Modal for Login.gov/Eligibility (Senior)
- [x] shorten radio button title to `Older Adult`
- [x] shorten radio button blurb
- [x] add new Login.gov logo image
- [x] add modal trigger (Login.gov image logo link) to the Older Adult radio button widget description
- [x] Create modal
- [x] ~Remove Help page link~ done in #1472 / #1517 
- Copy has to go into EligibiltyVerifier/Form model
- Login.gov logo must have a text-backup.
- Don't do the External Link icon yet.

## Modal trigger

Image

## Modal

Image

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #1473

### CODE REVIEW

**Comment by angela-tran** [comment]
There's some inconsistent behavior with closing the modal and what happens to the radio button.

See this GIF where I close the modal by:
 - clicking the X
 - pressing the Esc key
 - clicking off the modal

In the last case, the radio button gets selected.
benefits-login-gov-help-modal

I'm really close to implementing a fix to make it all consistent (as in, do not select the radio button), but I can't get the CSP nonce value into the `script` tag 😞 

```javascript
    <script nonce="{{ request.csp_nonce }}">
        const modal = document.getElementById("login-gov-help");
        modal.addEventListener('click', function (event) {
            event.preventDefault();
          });
    </script>
```

**Comment by angela-tran** [comment]
Renamed the modal file

**Comment by thekaveman** [comment]
> I'm really close to implementing a fix to make it all consistent (as in, do not select the radio button), but I can't get the CSP nonce value into the script tag 😞

Hmm this is a bummer 🤔 

I wonder if this is because you're inside an include and `request.csp_nonce` isn't available there? 

Have you tried the [`context_processor` version of `{{ CSP_NONCE }}`](https://django-csp-test.readthedocs.io/en/latest/nonce.html#context-processor)?

**Comment by angela-tran** [comment]
> Have you tried the [`context_processor` version of `{{ CSP_NONCE }}`](https://django-csp-test.readthedocs.io/en/latest/nonce.html#context-processor)?

Yeah, it's the same situation (not available in the include).

**Comment by thekaveman** [comment]
> An included template is rendered within the context of the template that includes it.

From https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#include

Lame, seems like it should work??

File: benefits/core/templates/core/includes/selection-label--senior.html
Comment (thekaveman) [inline benefits/core/templates/core/includes/selection-label--senior.html:1]: Side comment: I think we'll want this file (and the other selection labels) in the `eligibility` app

**Comment by angela-tran** [comment]
Hmm, it might have to do with this being a form widget template.

When rendering the `form` (which is an include), we still have the context:
image

We lose the context once we're rendering the widget.

image

**Comment by angela-tran** [comment]
Someone made a helper library for this very purpose: https://github.com/dmptrluke/django-csp-helpers#form-widgets 👀

**Comment by thekaveman** [comment]
Also this SO post from 2013! https://stackoverflow.com/a/16922285/453168

**Comment by angela-tran** [comment]
> Also this SO post from 2013! https://stackoverflow.com/a/16922285/453168

Such profit!

I propagated the `csp_nonce` through to the widget in a3854dff0ff012e2875ddfecfd66fba0617efe51. Let me know if that looks ok or not.

File: benefits/static/css/styles.css [hunk 1]
```diff
@@ -371,6 +374,10 @@ footer .footer-links li a:visited {
   line-height: 1;
 }
```
Comment (thekaveman) [inline benefits/static/css/styles.css:375]: I know this PR didn't introduce it, but I think we can fix a small bug in the Login.gov button logo.

We should add `background-repeat: no-repeat;` to this section of rules for the fallback text.

## Without this fix:

Here it is in `dev` right now in Chrome at tablet width showing the repeated image (appears like the shield is underlined):

image

Here it is from this PR, the modal button, where it is much more obvious in color:

image

## With the fix

Add one rule:

```css
#login .fallback-text {
  background-size: contain;
  background-repeat: no-repeat; /* <-- this one */
 /* ... */
}
```

Solves the issue on all buttons:

image

**Comment by thekaveman** [comment]
>> Also this SO post from 2013! https://stackoverflow.com/a/16922285/453168

> Such profit!

🤣

File: benefits/static/css/styles.css (hunk 1 above)
Comment (thekaveman) [reply benefits/static/css/styles.css:376]: Ah, I think it was right here from yesterday, snuck by: https://github.com/cal-itp/benefits/pull/1508/files#diff-744df555aee38173a0b0868baa7befa720e6a28f80747523e74e724c126cd381

Comment (angela-tran) [reply benefits/static/css/styles.css:376]: Oh I see. Thanks for catching that!

File: benefits/eligibility/templates/eligibility/includes/selection-label--senior.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <button id="login" type="button" class="modal-trigger--link" data-bs-toggle="modal" data-bs-target="#login-gov-help">
+        <span class="fallback-text color-logo">Login.gov</span>
+    </button>
+
+    {% include "eligibility/includes/modal--senior-help.html" with id="login-gov-help" %}
+
+    <script nonce="{{ widget.csp_nonce }}">
+        const modal = document.getElementById("login-gov-help");
+        modal.addEventListener("click", function (event) {
+            event.preventDefault();
```
Comment (thekaveman) [inline benefits/eligibility/templates/eligibility/includes/selection-label--senior.html:21]: Oof, unfortunately this is breaking the link, which is also processed as a `click` event.

Let's remove this and file an issue to follow-up.

Comment (thekaveman) [inline benefits/eligibility/templates/eligibility/includes/selection-label--senior.html:21]: Oof, unfortunately this is breaking the link, which is also processed as a `click` event.

Let's remove this and file an issue to follow-up.

**Review by thekaveman — CHANGES_REQUESTED** [review]

File: benefits/eligibility/templates/eligibility/includes/selection-label--senior.html (hunk 2 above)
Comment (angela-tran) [reply benefits/eligibility/templates/eligibility/includes/selection-label--senior.html]: I reverted the commit and created #1526

File: benefits/static/css/styles.css [hunk 3]
```diff
@@ -371,6 +375,10 @@ footer .footer-links li a:visited {
...
 
+.eligibility-index #login .fallback-text {
+  display: inline-block;
```
Comment (thekaveman) [inline benefits/static/css/styles.css:379]: This is so the `?` icon stays on the same line as the Login.gov image in mobile.

Maybe look at refactoring this rule instead (to not be inside a media query): https://github.com/cal-itp/benefits/blob/dev/benefits/static/css/styles.css#L385

**Review by thekaveman — APPROVED** [review]

**Pull request merged**